fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/DepositAccountWritePlatformServiceJpaRepositoryImpl.java

```diff
@@ -1324,8 +1324,15 @@ public void updateMaturityDetails(Long depositAccountId, DepositAccountType depo
                         fdAccount.getOnAccountClosureId(), toSavingsId, "Apply maturity instructions", changes);
 
                 if (changes.get("reinvestedDepositId") != null) {
+
                     Long reinvestedDepositId = (Long) changes.get("reinvestedDepositId");
+
                     Money amountForDeposit = account.activateWithBalance();
+                    if (((FixedDepositAccount) account).getAccountTermAndPreClosure().getOnAccountClosureType() == 300) {
```

Comment:
this could be broken down to check for null 
AccountTermAndPreClosure termAndPreClosure = fixedAccount.getAccountTermAndPreClosure();

Reply:
The filter on this getAccountTermAndPreClosure is up. This account should be activate to be checked here and that object should not be null. If it's null it's a bug